File: 🟢 Easy/1957 - Delete Characters to Make Fancy String/_1957DeleteCharacterstoMakeFancyString.cpp
class Solution {
public:
    string makeFancyString(string s) {
        char prev = s[0];
        int count = 1;

        string res = "";
        res.push_back(s[0]);

        for (int i = 1; i < s.size(); i++) {
            if (s[i] == prev) {
                count++;
            } else {
                count = 1;
                prev = s[i];
            }

            if (count < 3) {
                res.push_back(s[i]);
            }
        }

        return res;
    }
};

File: 🟢 Easy/1957 - Delete Characters to Make Fancy String/_1957DeleteCharacterstoMakeFancyString.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        prev = s[0]
        count = 1
        res = s[0]

        for i in range(1, len(s)):
            if s[i] == prev:
                count += 1
            else:
                count = 1
                prev = s[i]

            if count < 3:
                res += s[i]

        return res
